parser/Builder.cc

```diff
@@ -940,7 +940,17 @@ class Builder::Impl {
     }
 
     unique_ptr<Node> symbol_compose(const token *begin, sorbet::parser::NodeVec parts, const token *end) {
-        return make_unique<DSymbol>(collectionLoc(begin, parts, end), std::move(parts));
+        if (collapseStringParts(parts)) {
```

Comment:
I didn't understand motivation for this change

Reply:
`whitequark/parser` emits a string `"foo"` as `s(:str, "foo")`, but sorbet/parser emits `s(:dstr, s(:str, "foo"))`. Same for `xstr` by the way.

This `dstr` node doesn't have any useful information in its source maps (because there's simply no additional context that you could add, `s(:str)` has everything you need).

But I ported it to match `whitequark/parser`'s structure, probably I could do that via adding a few more logic to the rewriter, but I think it's not worth it.